Corgibytes.Freshli.Cli/Functionality/Analysis/AnalysisLocation.cs

```diff
@@ -13,13 +13,22 @@ public AnalysisLocation(IConfiguration configuration, string repositoryId, strin
         CommitId = commitId;
     }
 
+    public string? LocalDirectory { get; init; }
+
     public string RepositoryId { get; }
     public string? CommitId { get; }
 
     public string Path
     {
         get
         {
+            if (LocalDirectory != null)
+            {
+                return CommitId == null
+                    ? LocalDirectory
+                    : System.IO.Path.Combine(_configuration.CacheDir, "histories", RepositoryId, CommitId);
+            }
+
```

Comment:
Is there a way to implement this that doesn't duplicate the logic on line 37?

Would this work?

```csharp
if (LocalDirectory != null)
{
    return LocalDirectory;
}
```

And then the rest of the method as it was before?

Reply:
Not entirely but I did manage it with that check. We should only return the local directory if the commit id is null and the local directory is not null. So it's two if statements in one now but should be fine. :D

```csharp
if (CommitId == null)
{
      if (LocalDirectory != null)
      {
          return LocalDirectory;
      }

      return System.IO.Path.Combine(_configuration.CacheDir, "repositories", RepositoryId);
}

return System.IO.Path.Combine(_configuration.CacheDir, "histories", RepositoryId, CommitId);
```